deploy-shopware.php

```diff
@@ -0,0 +1,275 @@
+            run('mkdir -p {{shopware_public_path}}/recovery/install/data');
+            run('touch {{shopware_public_path}}/recovery/install/data/install.lock');
+        }
+    }
+)->desc('Check if a Shopware update is needed and execute it');
+
+task(
+    'kellerkinder:vendors',
+    function () {
+        foreach ([
+                     'k10r/staging:1.0.0',
+                     'k10r/deployment:1.1.0',
+                 ] as $dependency) {
+            run(
+                "cd {{shopware_public_path}} && {{bin/composer}} require --optimize-autoloader --prefer-dist --no-ansi --update-no-dev {$dependency}"
+            );
+        }
+    }
+)->desc('Add required composer-based Shopware plugins for deployment')->setPrivate();
+
+
+task(
+    'deploy:vendors',
+    function () {
+        run('cd {{shopware_public_path}} && {{bin/composer}} {{composer_options}}');
+    }
+)->desc('Install composer dependencies');
+
+task(
+    'kellerkinder:shopware:plugins',
+    function () {
+        run('cd {{shopware_public_path}} && {{bin/php}} bin/console sw:plugin:refresh -q');
+        run('cd {{shopware_public_path}} && {{bin/php}} bin/console sw:plugin:reinstall K10rDeployment');
+        run('cd {{shopware_public_path}} && {{bin/php}} bin/console k10r:plugin:deactivate SwagUpdate');
+
+        foreach (get('plugins') as $plugin) {
+            run(
+                "cd {{shopware_public_path}} && {{bin/php}} bin/console k10r:plugin:install --activate {$plugin}"
+            );
+        }
+    }
+)->desc('Install Shopware plugins after deployment. Configure plugins via deploy.php.');
+
+task(
+    'kellerkinder:shopware:config',
+    function () {
+        run("cd {{shopware_public_path}} && {{bin/php}} bin/console k10r:snippets:update");
+    }
+)->desc('Update snippets and set Shopware core configuration');
+
+
+task(
+    'kellerkinder:shopware:theme',
+    function () {
+        run(
+            "cd {{shopware_public_path}} && {{bin/php}} bin/console sw:theme:synchronize -q"
+        );
+        foreach (get('theme_config') as $setting => $value) {
+            run(
+            /**
+             * Adjust YourCustomTheme to your theme name
+             */
+                "cd {{shopware_public_path}} && {{bin/php}} bin/console k10r:theme:update -q --theme YourCustomTheme --shop 1 --setting {$setting} --value {$value}"
+            );
+        }
+    }
+)->desc('Configure theme via deploy.php');
+
+task(
+    'kellerkinder:shopware:cache',
+    function () {
+        run('cd {{shopware_public_path}} && {{bin/php}} bin/console sw:cache:clear -q');
+        run('cd {{shopware_public_path}} && {{bin/php}} bin/console k10r:theme:compile -q');
+        /**
+         * Warming the cache may take some time therefore it is not enabled by default. Enable if you wish to warm up the cache after every deployment.
+         */
+//        run('cd {{shopware_public_path}} && {{bin/php}} bin/console sw:warm:http:cache -c -q');
```

Comment:
Maybe we should use a boolean deployer setting around this.

Reply:
Good point. Fixed.